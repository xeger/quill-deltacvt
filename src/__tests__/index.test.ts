import { IMAGES, LISTS, MALICIOUS, SMORGASBORD } from './fixtures';
import * as visual from './visual';

import { generate } from '../index';

describe('generate', () => {
  it('handles images', () => {
    visual.matchSnapshot(generate(IMAGES));
  });

  it('escapes HTML syntax', () => {
    visual.matchSnapshot(generate(MALICIOUS));
  });

  it('handles lists', () => {
    visual.matchSnapshot(generate(LISTS));
  });

  it('handles standard Quill formats', () => {
    visual.matchSnapshot(generate(SMORGASBORD));
  });
});
