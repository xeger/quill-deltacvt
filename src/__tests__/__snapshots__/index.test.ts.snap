// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate escapes HTML syntax 1`] = `
"<div style=\\"font-family: sans-serif; white-space: pre-wrap\\">&lt;script&gt;alert(&quot;xss&quot;);&lt;/script&gt;
<ul style=\\"margin: 0px; list-style-position: inside; padding-left: 1.5rem\\"><li>&lt;script&gt;alert(&quot;xss&quot;);&lt;/script&gt;
</li><li style=\\"text-align: right;\\">&lt;script&gt;alert(&quot;xss&quot;);&lt;/script&gt;
</li><li>not xss &amp; okay
</li><li style=\\"text-align: center;\\">not xss &amp; okay

</li></div>"
`;

exports[`generate handles images 1`] = `
"<div style=\\"font-family: sans-serif; white-space: pre-wrap\\"><img src=\\"https://quilljs.com/assets/images/logo.svg\\" width=\\"140\\" height=\\"38\\"/>
Sized image.

<img src=\\"https://raw.githubusercontent.com/quilljs/quill/master/assets/favicon.png\\"/>
Left-aligned image.

<div style=\\"text-align: center\\"><img src=\\"https://raw.githubusercontent.com/quilljs/quill/master/assets/favicon.png\\"/>
Center-aligned image.

</div><div style=\\"text-align: right\\"><img src=\\"https://raw.githubusercontent.com/quilljs/quill/master/assets/favicon.png\\"/>
Right-aligned image.
</div>
<div style=\\"text-align: center\\"><img src=\\"https://quilljs.com/assets/images/logo.svg\\" width=\\"140\\" height=\\"38\\"/>
Sized and centered image.
</div>"
`;

exports[`generate handles lists 1`] = `
"<div style=\\"font-family: sans-serif; white-space: pre-wrap\\">Normal text to start.
<ul style=\\"margin: 0px; list-style-position: inside; padding-left: 1.5rem\\"><li>chalk
</li><li>
</li><li>cheese
</li></ul>Now an ordered list:
<ol style=\\"margin: 0px; list-style-position: inside; padding-left: 1.5rem\\"><li>apples
</li><li style=\\"text-align: right;\\">oranges
</li><li>pears
</li><li>
</li></ol>And finally, adjacent lists:
<ul style=\\"margin: 0px; list-style-position: inside; padding-left: 1.5rem\\"><li>please
</li><li>excuse
</li><li>my
</li></ul><ol style=\\"margin: 0px; list-style-position: inside; padding-left: 1.5rem\\"><li>dear
</li><li>aunt
</li><li>sally
</li></div>"
`;

exports[`generate handles standard Quill formats 1`] = `
"<div style=\\"font-family: sans-serif; white-space: pre-wrap\\">Plain.  <b>Bold. </b><i>Italic. </i><u>Underline.</u> <span style=\\"color: #9933ff\\">Colored.</span> <a href=\\"https://quilljs.com\\">Link.</a>
<span style=\\"font-size: x-small\\">Small, </span>normal, <span style=\\"font-size: x-large\\">large, </span><span style=\\"font-size: xx-large\\">huge!</span>
Left aligned.
Left <b>aligned</b> with <u><i>some</i></u> <span style=\\"color: #e60000\\">formatting</span>.
<div style=\\"text-align: center\\">Centered.
Centered <span style=\\"color: #b2b200\\"><b>with</b></span> <u><a href=\\"http://quilljs.com\\">some</a></u> formatting.
</div><div style=\\"text-align: right\\">Right aligned.
<span style=\\"color: #66b966\\">Right</span> <u><i>aligned</i></u> with <b>some</b> formatting.

</div><div style=\\"text-align: justify\\"><b>Justified</b> so the text nicely fills the line. <span style=\\"color: #f06666\\">Justified</span> <span style=\\"color: #66a3e0\\">so</span> <span style=\\"color: #c285ff\\">the</span> text nicely fills the line. Justified so the text nicely fills the line.
Also justified.
</div>
<ul style=\\"margin: 0px; list-style-position: inside; padding-left: 1.5rem\\"><li>Unordered
</li><li>List
</li></ul>
<ol style=\\"margin: 0px; list-style-position: inside; padding-left: 1.5rem\\"><li>Ordered
</li><li>List
</li></ol>
<ul style=\\"margin: 0px; list-style-position: inside; padding-left: 1.5rem\\"><li>Mixed
</li></ul><ol style=\\"margin: 0px; list-style-position: inside; padding-left: 1.5rem\\"><li>List
</li></ol><ul style=\\"margin: 0px; list-style-position: inside; padding-left: 1.5rem\\"><li>Weirdly
</li></ul><ol style=\\"margin: 0px; list-style-position: inside; padding-left: 1.5rem\\"><li>Laid Out
</li><li>With Numbers
</li></ol><ul style=\\"margin: 0px; list-style-position: inside; padding-left: 1.5rem\\"><li>Interspersed
</li></ul>
<ul style=\\"margin: 0px; list-style-position: inside; padding-left: 1.5rem\\"><li>list
</li><li style=\\"text-align: center;\\">with
</li><li style=\\"text-align: right;\\">alignment
</li><li style=\\"text-align: justify;\\">and justification in case the list items get stupendously enormously long
</li></div></ul>
</div>"
`;
